discovery-provider/src/queries/get_cid_source.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import sqlalchemy
+
+from src import exceptions
+from src.utils import db_session
+
+logger = logging.getLogger(__name__)
+
+
+def get_cid_source(cid):
+    """
+    Returns the CID source (e.g. CID is a metadata hash, a cover photo, a track segment, etc.)
+
+    Args: the observed CID
+    """
+    if cid is None:
+        raise exceptions.ArgumentError("Input CID is invalid")
+
+    db = db_session.get_db_read_replica()
+    with db.scoped_session() as session:
```

Comment:
this query seems very db heavy. we should at the very least add a lock to make sure only one instance of this is running now if we can't optimize it

Reply:
discussed offline - looking into locking mechanisms